packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md

```diff
@@ -0,0 +1,192 @@
+
+- The hooks pattern for controlled inputs is not very common in ecosystem. Didn't find this being used anywhere apart from downshift-js (in their case, everything is hook). Most popular libraries follow something along the lines of `onChange` + `isSelected`.
+- Initial assumption was that there will be too much of state management on user's end with `onChange` + `isSelected` type of approach but now that I look at the [overall example](#onchange--isselected-), it doesn't seem too complex. It also covers most usecases I could think of like `reset`, reading currently selected options, etc.
+- Additional learning curve of `useSelect` API. Explaining `isSelected` is a lot more easier.
+- Since `useSelect` will return `selectOption` method, consumers have to hook this into their state management.
+
+```jsx
+const { values } = useFormik();
+const { selectOption } = useSelect();
+
+const doSomethingComplex = () => {
+  // ... bunch of code that updates `values`
+  selectOption(values.status);
+};
+
+// As opposed to in `onChange` it will just look like this -
+const { values } = useFormik();
+
+<ActionListItem isSelected={values.status === 'pending'} title="Pending" />;
+```
+
+### Referrences
+
+- Primer
+  - [ActionMenu](https://primer.style/react/ActionMenu#with-selection) (Closest to our approach. Uses `selected` and `onSelect`)
+    ```jsx
+    <ActionList.Item
+      key={index}
+      selected={index === selectedIndex}
+      //
+      onSelect={() => setSelectedIndex(index)}
+    />
+    ```
+- Radix [[CodeSandbox Example](https://codesandbox.io/p/sandbox/holy-cache-5kwcgm?selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A28%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A24%7D%5D&file=%2FApp.jsx)]
+- [Polaris](https://polaris.shopify.com/components/selection-and-input/autocomplete) (They have `selected` object with `onSelect` on top component itself)
+- [downshift-js](https://github.com/downshift-js/downshift/tree/master/src/hooks/useSelect#actions) (`useSelect` hook)
+
+## Open Questions
+
+### Are `<ActionListItem />` "selectable" components?
+
+**Context:** We support `isDefaultSelected` on ActionList and plan to support `isSelected` now so that means we are treating ActionList as selectable items. When we have `Button` as trigger, we turn ActionList into menu which is not "selectable" but rather a list of "clickable" items.
+
+- **Pros of keeping ActionList selectable**
+
+  - Selection logic is still applicable in `<AutoComplete />`, or any selectable trigger we plan to add in future
+  - Here we are only talking about "selection" logic on consumer-facing props but selection logic on our end will anyway always be there irrespective of this call as we have single component that acts as `<MenuOption />` and `<SelectOption />`, both. Normally I have seen libraries tackle this in separate components.
+
+- **Cons**
+  - Not applicable in Button trigger so `isSelected`, `isDefaultSelected` would do nothing in that case
```

Comment:
It is definitely a con but we would have to take this tradeoff either way if we only have single `<ActionListItem />` component. 

Think of this in `SelectInput` where `<ActionListItem />`, rather than just being selectable items are trying to do some more actions. As a UX, this is not expected from SelectInput. It's more of a "menu" thing. So it not only impacts `isDefaultSelected` type of props but also impacts `onClick`, `href` on ActionListItem.
```jsx
<Dropdown>
  <SelectInput />
  <DropdownOverlay>
    <ActionList>
       <ActionListItem href="/settings" title="Settings" />
      <ActionListItem onClick={() => logOut()} title="Log Out" />
    </ActionList>
   </DropdownOverlay>
</Dropdown>
```

If we are treating `<ActionListItem />` as both menu and selectable item, there will always be props that work in menu but don't work in select and that work in select but don't work in menu since they are both UX-wise different items. 

If want to remove these impossible scenarios, we can create `<SelectOptions>` and `<SelectOptionItem>` kind of component that replace ActionList in Select. Then the selection logic goes into these components and ActionListItem handles menu items.

Although I still feel our API is good since consumers get one component that can do both so they don't have to think about which type of menu to use. But yup, it comes with these confusing impossible states between menu and select.